### DISCUSSION TITLE
Question on the implementation of `calc_waveform_metrics`

### DISCUSSION THREAD

**Discussion by oaaij-gnahz** [petersenpeter/CellExplorer#90] [Q&A] [answered]
Hi!

I was reading the implementation of `calc_waveform_metrics`, and on this line:
https://github.com/petersenpeter/CellExplorer/blob/master/calc_CellMetrics/calc_waveform_metrics.m#L24

It seems to be using a spline interpolation to adjust the sample timestamps of to-be-resampled waveform data.

However, it is simply interpolating the curve $y=x$ (the first two parameters passed to the function `interp1` are the same `timeWaveform`). 
I wonder in this case why would an interpolation or spline interpolation be needed? What is the consideration?

Thank you!

**Comment by petersenpeter** [comment, marked as answer]
Hi 

Thanks for pointing this out. Indeed it was not necessary - it was a lazy implementation on my part. I am interpolating the waveforms using the same method a few lines below, so I had simply reused the line for the time-axis, which was not necessary. I altered the line to the following: 
`timeWaveform = timeWaveform(1):mean(diff(timeWaveform))/oversampling:timeWaveform(end); `

I pushed the changes to GitHub.

**Reply by oaaij-gnahz** [reply]
Thanks for the clarification!
